src/main/java/de/medizininformatikinitiative/torch/service/DirectResourceLoader.java

```diff
@@ -153,14 +162,25 @@ private Mono<PatientBatchWithConsent> processPatientSingleAttributeGroup(Annotat
         return groupQueries(group)
                 .concatMap(query -> executeQueryWithBatch(batch.patientBatch(), query))
                 .concatMap(resource -> applyConsent(resource, batch))
-                .filter(resource -> !resource.isEmpty())
+                .filter(resource -> resourceBelongsToBatch(batch, resource))
+                .doOnDiscard(DomainResource.class, resource -> {
+                    if (resource.isEmpty()) {
+                        logger.warn("Ignoring empty resource {}", resource.getId());
```

Comment:
In case the resource is empty, it will not have an ID, right? If I think about it, is there any value in handling the empty case special? I mean if the resource has no patient-id it is skipped anyway. However the other direction isn't true, if the resource isn't empty, it's still possible that it has no patient-id. So I would remove the empty check from this, the logging part and from the filter.

Reply:
The empty case is also not covered by tests. I think we should start looking at code coverage.